tutorials/radiant-mlhub-on-demand-training-data.ipynb

```diff
@@ -0,0 +1,1281 @@
+   "id": "d7a5e313-0a2f-410e-8339-9630ca87dd78",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "l8_cropped[0].plot(col=\"band\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bc46c85b-daa9-44df-9be7-62dfa1234b25",
+   "metadata": {},
+   "source": [
+    "Now we have a cropped Landsat 8 chip that spatially and temporally matches our Sentinel-2 source imagery and label sample from the BigEarthNet dataset."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3a70b063-d273-4aee-864b-07e318388890",
+   "metadata": {},
+   "source": [
+    "### Launch a Dask gateway cluster for parallel processing"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cb650d80-bf1a-4aad-8f8b-08a612e28aae",
+   "metadata": {},
+   "source": [
+    "We will use Dask to optimize our data processing of hundreds of Landsat-8 scenes by parallelizing the workflow with a delayed computation graph. The Dask Client schedules, runs the delayed computations, and gathers the results, while the Dask Gateway provides a secure and centralized way of managing the multiple client clusters. This is especially useful for running Dask on Planetary Computer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "29531759-6d19-4010-8401-eb947a32c515",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client = dd_client(\n",
+    "    # you can configure Dask client parameters here\n",
+    ")\n",
+    "client"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "927a0496-a6c0-4aa2-9181-708904a5bb4f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# this cell will only work on PC or a machine with gateway cluster configured\n",
+    "# gateway = dask_gateway.Gateway()\n",
+    "# options = gateway.cluster_options()\n",
+    "# options[\"worker_cores\"] = 7"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f3b09697-b6ab-4026-bfcb-2f5214b03f5c",
+   "metadata": {},
+   "source": [
+    "### Scale the workflow using Dask Delayed"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "5368c39f-94a1-41ba-acc0-fd18c8dc1c18",
+   "metadata": {},
+   "source": [
+    "These are two helper functions that we will use to encapsulate the process of creating the cropped Landsat 8 chips and write them to disk in parallel using the Dask Delayed decorator."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c924acb1-092e-4f86-b73f-5b56ccdebe27",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def create_landsat_8_dataarray(item_path: str) -> DataArray:\n",
+    "    \"\"\"Creates a Landsat 8 chip from BigEarthNet label chip.\n",
+    "\n",
+    "    Args:\n",
+    "        item_path: string path to the label item on disk\n",
+    "\n",
+    "    Returns:\n",
+    "        Landsat 8 DataArray that has been cropped to label bbox\n",
+    "    \"\"\"\n",
+    "    # read label Item object\n",
+    "    label_item = Item.from_file(\n",
```

Comment:
This, and the other functions touch the local filesystem won't work with a distributed cluster. They'd only work with a LocalCluster.

If we're mentioning possibly using a distributed cluster then you'd need to restructure this. Most likely, you'd need to store everything to Azure Blob Storage and use it as a kind of shared file system that each worker could read / write to.

Reply:
For purposes of this tutorial, seems to make sense to focus on the workflow of gathering and processing the Landsat 8 data off of Planetary Computer with a `LocalCluster`, and not convoluting that with performing this workflow on a distributed cluster.